core/src/main/scala/kafka/utils/KafkaScheduler.scala

```diff
@@ -107,7 +107,10 @@ class KafkaScheduler(val threads: Int,
     debug("Scheduling task %s with initial delay %d ms and period %d ms."
         .format(name, TimeUnit.MILLISECONDS.convert(delay, unit), TimeUnit.MILLISECONDS.convert(period, unit)))
     this synchronized {
-      ensureRunning()
+      if (!isStarted) {
+        info("Kafka scheduler is not running at the time '%s' is scheduled.".format(name))
+        return null
```

Comment:
In UnifiedLog, we have code uses the returned future.

`  val producerExpireCheck = scheduler.schedule(name = "PeriodicProducerExpirationCheck", fun = () => {
`
With this change, perhaps we could return an Option and let the caller deal with it accordingly?